File: scripts/config-loader.js
import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

// Leer la configuración centralizada
const configPath = path.join(__dirname, '..', 'config.json');
const configData = JSON.parse(fs.readFileSync(configPath, 'utf8'));

// Determinar el entorno (development por defecto)
const environment = process.env.NODE_ENV || 'development';
const envConfig = configData[environment];

// Configuración del backend
export const BACKEND_CONFIG = {
  IP: envConfig.backend.ip,
  PORT: envConfig.backend.port.toString(),
  PROTOCOL: envConfig.backend.protocol,
  HOST: envConfig.backend.host
};

// Configuración del frontend
export const FRONTEND_CONFIG = {
  PORT: envConfig.frontend.port,
  HOST: envConfig.frontend.host,
  ALLOWED_HOSTS: envConfig.frontend.allowedHosts
};

// Configuración de la base de datos
export const DATABASE_CONFIG = {
  PORT: envConfig.database.port,
  HOST: envConfig.database.host
};

// Función para obtener la URL base del backend
export const getBaseUrl = () => {
  return `${BACKEND_CONFIG.PROTOCOL}://${BACKEND_CONFIG.IP}:${BACKEND_CONFIG.PORT}`;
};

// Configuración completa
export const config = {
  API_BASE_URL: getBaseUrl(),
  BACKEND_CONFIG,
  FRONTEND_CONFIG,
  DATABASE_CONFIG,
  ENVIRONMENT: environment
};

// Función para actualizar la configuración del backend
export const updateBackendConfig = (newConfig) => {
  Object.assign(BACKEND_CONFIG, newConfig);
  config.API_BASE_URL = getBaseUrl();
};

// Función para actualizar solo la IP
export const updateBackendIP = (newIP) => {
  BACKEND_CONFIG.IP = newIP;
  config.API_BASE_URL = getBaseUrl();
};

File: scripts/config_loader.py
import json
import os
from pathlib import Path

def load_config():
    """Cargar configuración centralizada desde config.json"""
    # Obtener la ruta del archivo de configuración
    current_dir = Path(__file__).parent
    config_path = current_dir.parent / "config.json"
    
    # Leer el archivo de configuración
    with open(config_path, 'r', encoding='utf-8') as f:
        config = json.load(f)
    
    # Determinar el entorno (development por defecto)
    environment = os.getenv('ENVIRONMENT', 'development')
    env_config = config.get(environment, config['development'])
    
    return env_config

def get_backend_config():
    """Obtener configuración específica del backend"""
    config = load_config()
    return config['backend']

def get_database_config():
    """Obtener configuración específica de la base de datos"""
    config = load_config()
    return config['database']

def get_frontend_config():
    """Obtener configuración específica del frontend"""
    config = load_config()
    return config['frontend']

# Configuración por defecto
_config = load_config()
BACKEND_CONFIG = _config['backend']
DATABASE_CONFIG = _config['database']
FRONTEND_CONFIG = _config['frontend']


